platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -0,0 +1,598 @@
+        scaleGestureDetector.onTouchEvent(event);
+        shoveGestureDetector.onTouchEvent(event);
+
+        // Handle two finger tap
+        switch (event.getActionMasked()) {
+            case MotionEvent.ACTION_DOWN:
+                // First pointer down
+                transform.setGestureInProgress(true);
+                break;
+
+            case MotionEvent.ACTION_POINTER_DOWN:
+                // Second pointer down
+                twoTap = event.getPointerCount() == 2
+                        && uiSettings.isZoomGesturesEnabled();
+                if (twoTap) {
+                    // Confirmed 2nd Finger Down
+                    MapboxEvent.trackGestureEvent(projection, MapboxEvent.GESTURE_TWO_FINGER_SINGLETAP, event.getX(), event.getY(), transform.getZoom());
+                }
+                break;
+
+            case MotionEvent.ACTION_POINTER_UP:
+                // Second pointer up
+                break;
+
+            case MotionEvent.ACTION_UP:
+                // First pointer up
+                long tapInterval = event.getEventTime() - event.getDownTime();
+                boolean isTap = tapInterval <= ViewConfiguration.getTapTimeout();
+                boolean inProgress = rotateGestureDetector.isInProgress()
+                        || scaleGestureDetector.isInProgress()
+                        || shoveGestureDetector.isInProgress();
+
+                if (twoTap && isTap && !inProgress) {
+                    if (focalPoint != null) {
+                        transform.zoom(false, focalPoint.x, focalPoint.y);
+                    } else {
+                        PointF focalPoint = TwoFingerGestureDetector.determineFocalPoint(event);
+                        transform.zoom(false, focalPoint.x, focalPoint.y);
+                    }
+                    twoTap = false;
+                    return true;
+                }
+
+                // Scroll / Pan Has Stopped
+                if (scrollInProgress) {
+                    MapboxEvent.trackGestureDragEndEvent(projection, event.getX(), event.getY(), transform.getZoom());
+                    scrollInProgress = false;
+                }
+
+                twoTap = false;
+                transform.setGestureInProgress(false);
+                break;
+
+            case MotionEvent.ACTION_CANCEL:
+                twoTap = false;
+                transform.setGestureInProgress(false);
+                break;
+        }
+
+        return gestureDetector.onTouchEvent(event);
+    }
+
+    /**
+     * Called for events that don't fit the other handlers.
+     * <p>
+     * Examples of such events are mouse scroll events, mouse moves, joystick & trackpad.
+     * </p>
+     *
+     * @param event The MotionEvent occured
+     * @return True is the event is handled
+     */
+    boolean onGenericMotionEvent(MotionEvent event) {
+        // Mouse events
+        //if (event.isFromSource(InputDevice.SOURCE_CLASS_POINTER)) { // this is not available before API 18
```

Comment:
it seems not.. after that line it indicates:  `// this is not available before API 18`, so it's  there waiting for us to bump the minSDK requirments? IDK, this code is from before my time.